sycl/test/basic_tests/marray/marray.cpp

```diff
@@ -23,11 +23,45 @@ using namespace sycl;
   CHECK_ALIAS_BY_SIZE(ALIAS_MTYPE, ELEM_TYPE, 8)                               \
   CHECK_ALIAS_BY_SIZE(ALIAS_MTYPE, ELEM_TYPE, 16)
 
+#define CHECK_BINOP(OP, LHS, RHS)                                              \
+  assert((LHS[0] * RHS) == (LHS * RHS) && (LHS * RHS[0]) == (LHS * RHS));
+
 struct NotDefaultConstructible {
   NotDefaultConstructible() = delete;
   constexpr NotDefaultConstructible(int){};
 };
 
+template <typename T> void CheckBinOps() {
+  sycl::marray<T, 3> ref_arr0{0};
+  sycl::marray<T, 3> ref_arr1{1};
+  sycl::marray<T, 3> ref_arr2{2};
+  sycl::marray<T, 3> ref_arr3{3};
+
+  CHECK_BINOP(+, ref_arr1, ref_arr2)
+  CHECK_BINOP(-, ref_arr1, ref_arr2)
+  CHECK_BINOP(*, ref_arr1, ref_arr2)
+  CHECK_BINOP(/, ref_arr1, ref_arr2)
+  CHECK_BINOP(&&, ref_arr0, ref_arr2)
+  CHECK_BINOP(||, ref_arr0, ref_arr2)
+  CHECK_BINOP(==, ref_arr1, ref_arr2)
+  CHECK_BINOP(!=, ref_arr1, ref_arr2)
+  CHECK_BINOP(<, ref_arr1, ref_arr2)
+  CHECK_BINOP(>, ref_arr1, ref_arr2)
+  CHECK_BINOP(<=, ref_arr1, ref_arr2)
+  CHECK_BINOP(>=, ref_arr1, ref_arr2)
+
+  if constexpr (!std::is_same_v<T, sycl::half> && !std::is_same_v<T, float> &&
+                !std::is_same_v<T, double>) {
```

Comment:
nit: I'd use early return instead here.

Reply:
Since the operations aren't defined for the given types, moving them out of the `if constexpr` would cause a compilation error, so we can't just do an early return here sadly, but it did help me catch a bug in the test!